### PR TITLE
chore(IT Wallet): [SIW-692] Add PIN auth during presentation

### PR DESCRIPTION
**Pull request by LazyAfternoons**
## Short description
This PR adds the PIN authentication while presenting claims. This happens in two flows: 
1. While issuing a credential, when presenting attributes required to obtain the credential;
2. While presenting a credential.

## List of changes proposed in this pull request
- Introduces a `itwSagaUtils.ts` which contains any generator function util which is used across sagas and adds a `verifyPin` utility;
- Replace existing pin authentications with `verifyPin`; 
- Adds `verifyPin` during the credential issuing flow and during presentation flows;

## How to test
1. Obtain a PID. PIN should be asked when storing the PID;
2. Obtain a credential. PIN should be asked when presenting claims required to obtain the credential and when storing the credential; 
3. Present your PID (scan QR code from the RP demo). PIN should be asked when confirming the presentation; 
4. Present your credential (long press on the credential). PIN should be asked when confirming the presentation.

### CODE REVIEW

File: ts/features/it-wallet/saga/ItwSagaUtils.ts
Comment (hevelius) [inline ts/features/it-wallet/saga/ItwSagaUtils.ts:1]: What do you think to use same file namecase convention?

File: ts/features/it-wallet/saga/ItwSagaUtils.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  identificationRequest,
+  identificationSuccess
+} from "../../../store/actions/identification";
+import I18n from "../../../i18n";
+
+/**
+ * Generator function that handles the PIN verification.
+ * Throws an error if the identification fails.
+ */
+export function* verifyPin() {
```
Comment (hevelius) [inline ts/features/it-wallet/saga/ItwSagaUtils.ts:15]: Is this an util function to add identification modal screen during a flow?

File: ts/features/it-wallet/saga/ItwSagaUtils.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      label: I18n.t("global.buttons.cancel"),
+      onCancel: () =>
+        NavigationService.dispatchNavigationAction(CommonActions.goBack())
+    })
+  );
+
+  const res = yield* take(identificationSuccess);
+
+  if (!isActionOf(identificationSuccess, res)) {
+    throw new Error(); // TO DO: needs to be mapped to an ITW error type in the future (SIW-713)
```
Comment (hevelius) [inline ts/features/it-wallet/saga/ItwSagaUtils.ts:27]:
```suggestion
    throw new Error(); // TODO: needs to be mapped to an ITW error type in the future (SIW-713)
```

File: ts/features/it-wallet/saga/itwRpPresentationSaga.ts [hunk 3]
```diff
@@ -25,61 +32,73 @@ export function* handleItwRpPresentationSaga(
...
-    ];
-
-    const requestObjectValue = yield* select(
-      itwRpInitializationRequestObjectValueSelector
+    yield* put(
+      identificationRequest(false, true, undefined, {
+        label: I18n.t("global.buttons.cancel"),
+        onCancel: () =>
+          NavigationService.dispatchNavigationAction(CommonActions.goBack())
+      })
```
Comment (hevelius) [inline ts/features/it-wallet/saga/itwRpPresentationSaga.ts:40]: What do you think to use verifyPin()?

**Review by hevelius — CHANGES_REQUESTED** [review]

File: ts/features/it-wallet/saga/ItwSagaUtils.ts (hunk 1 above)
Comment (LazyAfternoons) [reply ts/features/it-wallet/saga/ItwSagaUtils.ts:15]: Yes it is but in a more generic way it just verifies the user PIN in any saga.

File: ts/features/it-wallet/saga/itwRpPresentationSaga.ts (hunk 3 above)
Comment (LazyAfternoons) [reply ts/features/it-wallet/saga/itwRpPresentationSaga.ts:40]: This one slipped away, thank you. Addressed in bdbf0f0.

File: ts/features/it-wallet/saga/ItwSagaUtils.ts
Comment (LazyAfternoons) [reply ts/features/it-wallet/saga/ItwSagaUtils.ts:1]: Yes indeed, thanks. Addressed in bfc8c91.

**Review by hevelius — APPROVED** [review]
LGTM

**Pull request merged**